Src/Plugins/Security/SEC101_043.NuGetCredentialsValidator.cs

```diff
@@ -211,5 +214,48 @@ private static List<string> ExtractHostsHelper(string hostXmlAsString)
 
             return returnList ?? new List<string>();
         }
+
+        private static List<(string user, string password)> ExtractCredentials(string credentialXmlAsString)
+        {
+            var list = new List<(string user, string password)>();
+
+            try
+            {
+                var credentialsXml = new XmlDocument();
+                credentialsXml.LoadXml(credentialXmlAsString);
+
+                IEnumerable<XmlNode> credentials = credentialsXml?.ChildNodes[0]?.ChildNodes.Cast<XmlNode>();
+                foreach (XmlNode credential in credentials)
+                {
+                    string user = null;
+                    string password = null;
+                    foreach (XmlNode item in credential.ChildNodes.Cast<XmlNode>())
+                    {
+                        IEnumerable<XmlNode> current = item.Attributes.Cast<XmlNode>();
+                        if (current.Any(a => a.Value.Equals("username", StringComparison.OrdinalIgnoreCase)))
+                        {
+                            user = current.FirstOrDefault(a => a.Name.Equals("value", StringComparison.OrdinalIgnoreCase))?.Value;
+                        }
+
+                        if (current.Any(a => a.Value.Equals("password", StringComparison.OrdinalIgnoreCase) ||
+                                             a.Value.Equals("cleartextpassword", StringComparison.OrdinalIgnoreCase)))
+                        {
+                            password = current.FirstOrDefault(a => a.Name.Equals("value", StringComparison.OrdinalIgnoreCase))?.Value;
+                        }
+                    }
+
+                    if (!string.IsNullOrEmpty(user) && !string.IsNullOrEmpty(password))
+                    {
+                        list.Add((user, password));
+                    }
+                }
+            }
+            catch (Exception)
```

Comment:
>Exception [](http://example.com/codeflow?start=19&length=9)

You need to catch a more specific exception type here, e.g., XmlException may be appropriate. Otherwise you may swallow bugs that we don't have an easy way to detect.

I'd actually suggest wrapping the LoadXml call only in a try catch (again catching XmlException only).

Reply:
thanks for the suggestion!